server.js

```diff
@@ -0,0 +1,75 @@
+import mongoose from "mongoose";
+import dotenv from "dotenv";
+import express from "express"
+import bodyParser from "body-parser"
+import cors from "cors"
+import validator from 'validator';
+
+dotenv.config({ path: "./config.env" });
+
+const connectionString = process.env.MONGO_URL
+const database = await mongoose.connect(connectionString);
+const app = express();
+
+console.log(`Connected to ${connectionString}`);
+
+const userSchema = mongoose.Schema({ username: String, email: String, password: String, userUID: String, isInstructor: { type: Boolean, default: false } });
+const userModel = database.model("users", userSchema);
+const users = await userModel.find();
+
+
+const urlencodedParser = bodyParser.urlencoded({ extended: false });
+app.use(bodyParser.json(), urlencodedParser)
+app.use(cors())
+
+
+app.listen(process.env.PORT, () => {
+    console.log(`App is Listening on PORT ${process.env.PORT}`)
+})
+
+app.get("/", async (req, res) => {
+    const users2 = await userModel.find();
+    console.log(users2)
+    res.json({ message: "Success" })
+})
+
+//still need custom validator for uid, turned to string for now
+app.post("/user", async (req, res) => {
+    const user = req.body
+    const validEmail = validator.isEmail(user.userInfo.email)
+    const takenEmail = await userModel.findOne({ email: user.userInfo.email })
+    if (takenEmail) {
+        res.status(403).json({msg:'test'})
```

Comment:
"test" isn't a helpful message.  "User already exists (try logging in instead)" might be better.

Likewise, "Invalid e-mail address" and "Invalid password" might be better for the subsequent lines.